Library/Homebrew/cask/cask_loader.rb

```diff
@@ -356,18 +361,12 @@ def self.for(ref, need_path: false)
         FromInstanceLoader,
         FromContentLoader,
         FromURILoader,
+        FromAPILoader,
```

Comment:
Run before `FromTapLoader` to preserve the same intended behavior as before.

Reply:
Currently, if the main cask tap is specified but the cask doesn't exist from the API it will skip the `FromAPILoader` and will try the `FromTapLoader` which will throw an error.

```
~ [1]$ brew install -n homebrew/cask/1passwordless --cask --debug
/usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.6.20-148-gaa35cc5-dirty\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.7\)\ curl/7.64.1 --header Accept-Language:\ en --fail --progress-bar --max-time 5 --retry 3 --location --remote-time --output /Users/kevinrobell/Library/Caches/Homebrew/api/cask.json --time-cond /Users/kevinrobell/Library/Caches/Homebrew/api/cask.json --compressed --silent https://formulae.brew.sh/api/cask.json
Warning: Cask 'homebrew/cask/1passwordless' is unavailable.
Please tap it and then try again: brew tap homebrew/cask
```

Maybe it'd be better to just try to load all cask references that start with the `homebrew/cask/` in the `FromAPILoader` so that we can throw an error that doesn't suggest tapping the main cask tap (the same casks should be available from both places, right?). This is the current behavior though so I didn't want to change it without some discussion.

Reply:
> Maybe it'd be better to just try to load all cask references that start with the homebrew/cask/ in the FromAPILoader so that we can throw an error that doesn't suggest tapping the main cask tap (the same casks should be available from both places, right?).

This makes sense to me (but can wait until another PR if easier).

Reply:
In that case, this can be simplified even further.

Reply:
Not sure I'm brave enough to make that change in this PR. 😁